### ISSUE TITLE
Social Card 1: When sharing a member's vote for a particular policy.

### ISSUE THREAD

**Issue by tseyiokorodudu** [openaustralia/theyvoteforyou#1339]
### This issue is linked to issue #1316

This card is for when someone shares ["a member's vote on a particular policy" page](https://theyvoteforyou.org.au/people/senate/tasmania/eric_abetz/policies/162). 

This is the original design shared on issue #1316 
![](https://user-images.githubusercontent.com/31440427/153092295-799d20eb-f905-4f89-b384-a279c19db840.png)

Any further correspondence about this particular card would be done here.

**Comment by tseyiokorodudu** [comment]
> For design 1, I opted for black bold text as opposed to white because I realised that white against the coloured background was a little hard to read and also had a low accessibility contrast. 

In issue #1316 I explained why we chose the text color for the initial card design. I've attached an image using the black text on the coloured background and one using the white text so we can see what I mean by my comment. 


![100001_1-2](https://user-images.githubusercontent.com/31440427/153995820-e0ef6296-c5ef-46da-a779-eb9572ca1616.png)


![100001_1](https://user-images.githubusercontent.com/31440427/153995840-3796ffff-6d20-4ae9-a684-28a0a73d3832.png)

**Comment by mlandauer** [comment]
For anyone coming to this issue blind it's been split out from #1316. This issue just covers the design and implementation of one of the cards. #1316 covers several cards.

**Comment by mlandauer** [comment]
@tseyiokorodudu do you want me to give feedback on the versions in https://github.com/openaustralia/publicwhip/issues/1339#issuecomment-1039871092 or were they purely meant to demonstrate the contrast issue? I'm asking because it's not clear to me and there have been some other changes made.

**Comment by tseyiokorodudu** [comment]
@mlandauer Yeah I would like some feedback. Sorry for not making that clear.

**Comment by mlandauer** [comment]
Here's some more feedback in no particular order again:

* Looks clearer with "They Vote For You" at the top
* I'm guessing from the text that this is actual rendered content from the implementation? If so, great!
* The font looks a bit weird and jagged to me. Is it the same font as used on the site?
* I agree that the contrast is way too low with the white writing. It's annoying because we have the same issue on the landing page of the site. At least on the site it's absolutely giant writing, which I guess helps to make it somewhat readable even if it isn't sticking to any accessibility standard.
* Does the black need to be absolutely black (#000)? If feels quite harsh and unfriendly to my eye. On the site it's using rgb(68, 68, 68). I don't know if it's strictly relevant but this brings to mind this article: https://uxmovement.com/content/why-you-should-never-use-pure-black-for-text-or-backgrounds/
* The border around "They Vote For You" feels quite tight. The design of the site is quite spacious. That feels a bit cramped to me.
* The link at the bottom feels too prominent to me. The use of "SOURCE:" is a bit loud and geeky. What would be a more human way of saying that? Could it be, for instance, more like the watermark that you get in the bottom right of movies often?
* Random thought - please take it or leave it: What if the lighter coloured areas were bands across the top of the whole card rather than boxes that surround the text? This would be more in line with the menu design of the site.... Might be a terrible idea and you probably have already considered but just mentioning in case...

**Comment by tseyiokorodudu** [comment]
- Yeah, it is the actual rendered content. 
- It is the same font as that of the site. 

> Random thought - please take it or leave it: What if the lighter coloured areas were bands across the top of the whole card rather than boxes that surround the text? This would be more in line with the menu design of the site.... Might be a terrible idea and you probably have already considered but just mentioning in case...

I actually considered it but never implemented it because I wanted the card design to look a little different from the site. But I could do it and see how it looks. 

- I don't exactly get the movie watermark reference - do you mean place the link at the right hand corner and make the size a bit smaller?

**Comment by Taz17** [comment]
Here is an example of a watermark on a movie. 

Image source: https://www.plagiarismtoday.com/2016/05/10/using-dynamic-video-watermarks-prevent-plagiarism-piracy/
![image](https://user-images.githubusercontent.com/86769131/154186913-acd76311-f685-4911-b563-64ecb9245a93.png)

Its the opaque white text in the bottom left corner of this image.

**Comment by tseyiokorodudu** [comment]
Worked on the design a bit based on the feedback above. 

- I changed the font colour from pure black to #262626 on the "MP name & Policy" text and used the site's default text colour rgb(68, 68, 68) on the site name and link. Does it look better and more friendly to the eye?

- Regarding the border around the site name looking quite tight, I tried out two things;
1. Increase the surrounding space as seen below
![cardlogo](https://user-images.githubusercontent.com/31440427/154196762-a6546de8-142c-49a9-90de-5ada207803b2.png)

2. Made the lighter colour spread across the top of the card.
![across](https://user-images.githubusercontent.com/31440427/154197333-4f872e75-8bfa-4cdb-a2e5-71c1569f034c.png)

- I reduced the font size of the link a bit. However, I'm still struggling with where to place it and if it should have a coloured background or not. Personally, I think it needs a coloured background to make it clearer and readable if I'm sticking with the present font colour. Do any of the placements in the card designs above work?

@mlandauer @katska Any thoughts?

**Comment by katska** [comment]
Given that some of @mlandauer's comments related to matching the feel of the site, I scaled up the page in the browser to see what that looked like with big text. That's one lense.
It looks like this, which is pretty interesting as a comparison

<img width="829" alt="Screen Shot 2022-02-16 at 5 12 47 pm" src="https://user-images.githubusercontent.com/1001129/154207328-6c84740a-209b-4d40-87eb-35b75db6ebbd.png">

So if at this stage we're still working out what to match back to the webpage and then figure out what we need to change and depart from to make sense as a card, then maybe good to bring parts we're not sure about working back to how it is in the site, use that visual language and depart where necessary for the social card format as a next pass. ?

I observe
- The banner works for me in the context of it's like the website and gives a TVFY feel :)
- Text begining Eric Abetz....  lines up from top left next to image, rather than centred
- Name is part of a sentence rather than a separate line 
- Balance of font sizes across the card not quite working. There's a very different relationship with They Vote For You text on the site, which is bigger and bolder. That's tricky because we don't necessarily want to be so bold in a card. How about we see it as per website and go from there?
- The organisation of information on the card is currently more like [people page]( https://theyvoteforyou.org.au/people/senate/tasmania/eric_abetz) than [policy page](https://theyvoteforyou.org.au/people/senate/tasmania/eric_abetz/policies/162) so if we're following the visual language of the site, this replaces his job position with his position on an issue. That seems inconsistent. 
- There's a difference between font in website (on this page specifically) and the card. See capital E for clear example of this. On another page it appears differently, so maybe there's some inconsistency at TVFY that's thrown this problem up.

**Comment by katska** [comment]
The watermark might be a separate issue. Would it work to have a footer at the bottom, a different colour that carries the link to the information? 
A watermark is a different function, is not intended to be very readable, so while its an attractive idea, I don't know if its the right concept for this link. 
Visual watermarks are also embedded to prevent theft or distortion. Our watermark would not carry this kind of verifying function if sitting on a plain background.

The more I think about it the more I think it's right that the full link be very legible if it's so people can check the source.
However the full link is not a very useful address to someone who sees it in a rendered image that does not embed a link to the source and is not easy to remember and follow.

**Comment by mlandauer** [comment]
On the font issue it looks to me now clearly (agreeing with what Kat said above) that the font you're using is not the same as the font in the application.

In fact it looks like the correct font is not being loaded and it's falling back to the default sans-serif which I guess some kind of helvetica

## This is what the application looks like normally
![With correct font](https://user-images.githubusercontent.com/2221/154375720-57acc370-5208-4d5e-8970-747e1cc83a19.png)

## This is what happens when it doesn't load the font that it's supposed to and falls back to a default
![Screen Shot 2022-02-17 at 10 37 49](https://user-images.githubusercontent.com/2221/154375723-ec37d17f-c5cb-4381-bc30-308d81a57e16.png)

**Comment by mlandauer** [comment]
* I also agree with @katska about the font sizes not holding together. There's four different font sizes in the one image and that makes it feel a bit messy and not coherent. It's funny me saying that because planningalerts.org.au is a complete dog's breakfast for having multiple fonts and sizes all over the place. TheyVoteForYou is generally much more coherent largely due @equivalentideas's strong focus on typography in the early development.
* Now that @katska brings up the idea of quite literally putting the name and the policy sentence together as one it makes it obvious to me that the font sizes don't make sense here. "Eric Abetz" is not a more important concept than the stance he has on a particular policy. The two bits together have equal weighting. So, shouldn't they be the same size?
* The band across the top (like the nav on the site) does make a little more sense of the otherwise dead space on the top right but I'm certainly not wedded to it.
* With the new sensible decision to go with dark text for legibility would it make more sense to give with background colour scheme of https://theyvoteforyou.org.au/policies instead? 
* Consistency of alignment - also feels a bit weird having a mixture of left aligned and centred on a single small image.
* The link in its current form at the bottom isn't working for me. It's visually very distracting and busy. I understand the need to make it visible. Are there any other options you could try here?
* I'd like some extra words next to the link to make it a little more human. Maybe "Discover more at ..." or "More at ..." or "Learn more ..."?

**Comment by tseyiokorodudu** [comment]
Thanks for the feedback. Based on that we've done the following;

- Looked into the font issue and fixed the problem.
   
- Added a javascript code, that automatically resizes the text to fit its container for policies with longer name.
 
- Followed @katska suggestion and designed the card to look like the site, so the name and policy are now on the same line.

- Increased the font size of the site name a little bit.

- Based on @mlandauer suggestion, we changed the background colour to look like  https://theyvoteforyou.org.au/policies. 

### Card with Lemon light background as in https://theyvoteforyou.org.au/policies.
![100001_1_lemon_light](https://user-images.githubusercontent.com/31440427/154385790-c201f414-7d8e-4b5e-90dc-a8b12e7f1225.png)

### Card with Orange background as in Homepage
![100001_1_orange](https://user-images.githubusercontent.com/31440427/154385815-de874912-73c4-4dbe-99fd-b93e3927312e.png)

We are leaning towards the lemon-light background at the moment. It feels the card feel less clunky and breezy.

- Still working on the link.

**Comment by mlandauer** [comment]
Thanks for the update @tseyiokorodudu. It's starting to feel more coherent to me. It's interesting seeing the card with the lighter background. I agree it feels a little less clunky and more breezy with it. However, the blocks of the slightly darker colour around some of the text makes less sense. I do think now that darker background colour needs to be a large area for it to make sense. So, that makes me inclined to think that the band across the top is the way to go. What do you think?

Since you're still working through the link stuff I'll wait on something landing here before commenting further.

**Comment by tseyiokorodudu** [comment]
### Link Update

#### Some Thoughts
Firstly, I think we need the background. As the alternative would be just the plain text like in the image below which from my perspective looks a bit weird and out of place from the other elements on the card.

![image](https://user-images.githubusercontent.com/31440427/154416050-58eb1608-c9c6-47a7-a275-4e9635375e35.png)

Secondly, I put the link at the bottom because placing it in any other place doesn't really work, especially given its length. To explain further, If I put it at the top, it makes it look crowded. if I put it just below the main text (i.e Mp's vote on a policy), it would break into two lines ... and that doesn't look good either and makes it a bit difficult to read the link. 

Thirdly, I can't reduce the font-size smaller than what it is now for legibility and readability purposes.

#### Current Design
Above @katska asked this question and I decided to give it a try.
> Would it work to have a footer at the bottom, a different colour that carries the link to the information?

Here is the result.

**Image 1**
![Link-footer-right](https://user-images.githubusercontent.com/31440427/154417484-1014f1c9-b048-4822-95b2-3e389f54c8ff.png)

What do you think? I know the blue is eye catching but I don't think it is distracting from the main thing especially now that the main text has a bigger font-size.

@mlandauer I'm actually starting to agree with the band across the top idea, because I think it makes the design feel "whole"/more coherent. However, I'm not 100% sold on it with current link style . Below is sample of what it looks like with the current link style for reference.
 
**Image 2**
![Link-footer-white](https://user-images.githubusercontent.com/31440427/154418963-6a448b9a-8937-467d-a03a-f1b637bba670.png)

To conclude;

1. What do you think of the current link style?
2. Do you prefer band across the whole card (image 2) or band the length of site name (Image1)

**Comment by mlandauer** [comment]
> @mlandauer I'm actually starting to agree with the band across the top idea, because I think it makes the design feel "whole"/more coherent. However, I'm not 100% sold on it with current link style . Below is sample of what it looks like with the current link style for reference.

I'm also now realising the "band across the whole width" idea also gives more space to the text and makes the design feel less crowded. It also makes the text alignment more cleanly follow the same line on the left which seems right to me. I don't understand what you mean by "current link style" in your comment above. Do you mean the general visual styling of the words "They Vote For You" or something else?

I like that you've tried out the "standard" footer colour above. I do find the blue quite jarring here and feels a bit out of place and distracts me from the body of the image. It also doesn't feel very balanced with the top band which is a different size and colour with different font sizing. I really don't think the link deserves a similar amount of attention as the heading. The most important job of the card is to convey the body of the message "Eric Abetz has never...". An equal amount of importance is to the fact that the information came from "They Vote For You" whatever that is. If the person looking at this is interested enough in the body of the information then and only then will they be interested in going to the link in which case they'll use that. So the link in my view is less important than the other two things.

The example in https://user-images.githubusercontent.com/31440427/154416050-58eb1608-c9c6-47a7-a275-4e9635375e35.png without the background colour I actually don't find too strange. It seems kind of okay to me. It would be better IMHO in that example if the band across the top was used which would then have left aligned "They Vote For You" with the text of the link.

Another thought on the text wording what if it said "How is this worked out? https://.." - the how is this worked out answers a question people often have and is an extremely reasonable question. Also by putting it as a question we incite peoples' curiosity. We've found things phrased as questions can be particularly engaging on social media.

**Comment by mlandauer** [comment]
I think we're getting extremely close to this being "good enough" to open a PR so we can work through getting a first version for this particular page rolled out into production 🎉

**Comment by tseyiokorodudu** [comment]
By "current link style" I meant the visual styling of the url at the bottom which you've given feedback on.  Thanks.

So here is the current design of the card.

![100001_1-3](https://user-images.githubusercontent.com/31440427/154872371-54c63527-27ed-4b7b-a3f2-522b223efe0c.png)

**Comment by Taz17** [comment]
We also chose to remove the `https://` from the URL since a user would be able to visit the site without it.

**Comment by mlandauer** [comment]
@tseyiokorodudu this is looking good to me! 👍🏼 

What do you think?

**Comment by tseyiokorodudu** [comment]
It looks good to me too.

**Comment by mlandauer** [comment]
> It looks good to me too.

Great. @tseyiokorodudu could you put together a pull request then for this please

**Comment by tseyiokorodudu** [comment]
Since we have gotten card 1 to a good enough place for now, I think we can begin work on the next card. See issue #1349 for more.

**Comment by mlandauer** [comment]
These cards are now being rendered in production by a rake task that runs every night at around 2am.

@Taz17 @tseyiokorodudu could you now please put together a PR for these to be used when sharing the page on social media?

Thank you!
